examples/webgl_box_projected_cubemap_environment_mapping.html

```diff
@@ -0,0 +1,484 @@
+
+						vec3 reflectView = normalize( ( viewMatrix * vec4( reflectVec, 0.0 ) ).xyz + vec3( 0.0,0.0,1.0 ) );
+
+						#ifdef TEXTURE_LOD_EXT
+
+							vec4 envMapColor = texture2DLodEXT( envMap, reflectView.xy * 0.5 + 0.5, specularMIPLevel );
+
+						#else
+
+							vec4 envMapColor = texture2D( envMap, reflectView.xy * 0.5 + 0.5, specularMIPLevel );
+
+						#endif
+
+						envMapColor.rgb = envMapTexelToLinear( envMapColor ).rgb;
+
+					#endif
+
+					return envMapColor.rgb * envMapIntensity;
+				}
+				#endif
+				`;
+
+			// scene size
+			var WIDTH = window.innerWidth;
+			var HEIGHT = window.innerHeight;
+
+			// camera
+			var VIEW_ANGLE = 45;
+			var ASPECT = WIDTH / HEIGHT;
+			var NEAR = 1;
+			var FAR = 800;
+
+			var camera, scene, renderer;
+
+			var cameraControls;
+
+			var groundPlane, smallSphere, wallMat;
+
+			init();
+			animate();
+
+			function init() {
+
+				var container = document.getElementById( 'container' );
+
+				// renderer
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( WIDTH, HEIGHT );
+				container.appendChild( renderer.domElement );
+
+				// gui controls
+				var gui = new dat.GUI();
+				var params = {
+					'box projected': true
+				};
+				var bpcemGui = gui.add( params, 'box projected' );
+
+				bpcemGui.onChange( function( value ) {
+					if ( value ){
+
+						groundPlane.material = boxProjectedMat;
+
+					} else {
+
+						groundPlane.material = defaultMat;
+
+					}
+				});
+
+				// scene
+				scene = new THREE.Scene();
+
+				// camera
+				camera = new THREE.PerspectiveCamera( VIEW_ANGLE, ASPECT, NEAR, FAR );
+				camera.position.set( 280, 106, -5 );
+
+				cameraControls = new THREE.OrbitControls( camera, renderer.domElement );
+				cameraControls.target.set( 0, -10, 0 );
+				cameraControls.maxDistance = 400;
+				cameraControls.minDistance = 10;
+				cameraControls.update();
+
+				// cube camera for environment map
+				cubeCamera = new THREE.CubeCamera( 1, 1000, 512 );
+				cubeCamera.renderTarget.texture.generateMipmaps = true;
+				cubeCamera.renderTarget.texture.minFilter = THREE.LinearMipMapLinearFilter;
+				cubeCamera.renderTarget.texture.magFilter = THREE.LinearFilter;
+				cubeCamera.renderTarget.texture.mapping = THREE.CubeReflectionMapping;
+
+				cubeCamera.position.set( 0, -100, 0 );
+				scene.add( cubeCamera );
+
+				// ground floor ( with box projected environment mapping )
+				var loader = new THREE.TextureLoader();
+				var rMap = loader.load( 'textures/lava/lavatile.jpg' );
+				rMap.wrapS = THREE.RepeatWrapping;
+				rMap.wrapT = THREE.RepeatWrapping;
+				rMap.repeat.set(2,1);
+
+				var defaultMat = new THREE.MeshPhysicalMaterial({
+					roughness: 1,
+					envMap: cubeCamera.renderTarget.texture,
+					roughnessMap: rMap,
+				});
+
+				var boxProjectedMat = new THREE.MeshPhysicalMaterial({
+					color: new THREE.Color('#ffffff'),
+			    roughness: 1,
```

Comment:
There are some minor formatting issues in your file. Please ensure to use only hard tabs.